### PR TITLE
Consumer Posthook VarArgs

### PR DESCRIPTION
**Pull request by SentryMan**
The asymmetry is really starting to get to me

### CODE REVIEW

File: inject/src/main/java/io/avaje/inject/BeanScopeConsumer.java [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package io.avaje.inject;
+
+import java.util.function.Consumer;
+
+@FunctionalInterface
```
Comment (rob-bygrave) [inline inject/src/main/java/io/avaje/inject/BeanScopeConsumer.java:5]: Need some javadoc here?

Comment (SentryMan) [reply inject/src/main/java/io/avaje/inject/BeanScopeConsumer.java:5]: I mean the name is pretty descriptive

**Comment by Mechite** [comment]
Wouldn't this result in heap pollution (since its varargs with generics)?

**Comment by SentryMan** [comment]
nah, because we create a separate non-generic interface that extends `Consumer`.

**Comment by SentryMan** [comment]
> use @SafeVarargs

`@SafeVarargs annotation cannot be applied to non-final instance method addPostConstructHooks`

I'm pretty sure what I have already should work.

**Comment by SentryMan** [comment]
Tried on my local, and I get compiler warnings everywhere I use the postHook method, I think I'll stick with the interface approach

File: inject/src/main/java/io/avaje/inject/BeanScopeBuilder.java [hunk 2]
```diff
@@ -216,16 +216,16 @@ default <D> BeanScopeBuilder provideDefault(Type type, Supplier<D> provider) {
...
 
   /**
-   * Adds hook that will execute after this scope is built.
+   * Adds hooks that will execute after this scope is built.
    *
-   * @param consumer the PostConstruct hook to add to the bean scope
+   * @param consumers the PostConstruct hooks to run after the BeanScope is constructed
    */
-  BeanScopeBuilder addPostConstructConsumerHook(Consumer<BeanScope> consumer);
+  BeanScopeBuilder addPostConstructHooks(BeanScopeConsumer... consumers);
```
Comment (rbygrave) [inline inject/src/main/java/io/avaje/inject/BeanScopeBuilder.java:228]: I suspect this is only ever called with 1 consumer.  I suspect the same for the Runnable hooks.

... so I _think_ these should both be changed to not use var-args ?

Comment (SentryMan) [reply inject/src/main/java/io/avaje/inject/BeanScopeBuilder.java:228]: Oh yeah, we could also do that

**Pull request closed without merging**